goldfish/srfi/srfi-133.scm

```diff
@@ -21,14 +21,40 @@
   vector-count
   vector-any vector-every vector-copy vector-copy!
   vector-index vector-index-right vector-partition
-  vector-swap! vector-cumulate reverse-list->vector)
+  vector-swap! vector-cumulate reverse-list->vector
+  vector=)
 (begin
 
 (define (vector-empty? v)
   (when (not (vector? v))
     (error 'type-error "v is not a vector"))
   (zero? (vector-length v)))
 
+; (vector= elt=? vec ...) -> boolean, srfi-133 support. 
+; Use the given element comparators to compare the input vectors.
+; When the input is null or just one vector, return #t.
+; Else, it will compare the vectors in the argument list one by one.
+;     If all the vectors are equal under the elt's comparison, return #t, else #f.
+(define (vector= elt . rest)
+  ; The inner method provides type-check and two vectors comparison.
+  (define compare2vecs
+    (typed-lambda ((cmp procedure?) (vec1 vector?) (vec2 vector?))
+      (let*  ((len1 (vector-length vec1))
+              (len2 (vector-length vec2)))
+        (if (not (= len1 len2)) #f
+            (let loop ((ilhs 0) (irhs 0) (imax len1))
+                 (if (= ilhs imax) #t
+                     (if (not (cmp (vec1 ilhs) (vec2 irhs))) #f
+                         (loop (+ 1 ilhs) (+ 1 irhs) imax))))))))
+  (if (not (procedure? elt)) (error 'type-error "elt should be a procedure")
+      (if (or(null? rest) (= 1 (length rest))) #t
+          (let loop ((vec1 (car rest))
+                     (vec2 (car (cdr rest)))
+                     (vrest (cdr (cdr rest))))
+               (if (compare2vecs elt vec1 vec2)
```

Comment:
`(if 1 #t #f)` is `#t`

How about make sure the result of compare2vecs is boolean?